### PR TITLE
feat: add get quote data

### PR DESCRIPTION
**Pull request by robercano**
Added `getSwapQuote` to Swap service

### CODE REVIEW

File: apps/summerfi-api/lib/sdk-router-function/src/handlers/getSwapQuote.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { QuoteData } from '@summerfi/swap-common/types'
+
+export const getSwapQuote = publicProcedure
+  .input(
+    z.object({
+      chainInfo: z.custom<ChainInfo>((chainInfo) => chainInfo !== undefined),
+      fromAmount: z.custom<TokenAmount>((tokenAmount) => tokenAmount !== undefined),
+      toToken: z.custom<Token>((token) => token !== undefined),
+      recipient: z.custom<Address>((recipient) => recipient !== undefined),
+      slippage: z.custom<Percentage>((slippage) => slippage !== undefined),
```
Comment (piotrwitek) [inline apps/summerfi-api/lib/sdk-router-function/src/handlers/getSwapQuote.ts:13]: those are not used

File: apps/summerfi-api/lib/sdk-router-function/src/handlers/getSwapQuote.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    }),
+  )
+  .query(async (opts): Promise<QuoteData> => {
+    return await opts.ctx.swapService.getSwapQuote({
+      chainInfo: opts.input.chainInfo,
+      fromAmount: opts.input.fromAmount,
+      toToken: opts.input.toToken,
+    })
```
Comment (piotrwitek) [inline apps/summerfi-api/lib/sdk-router-function/src/handlers/getSwapQuote.ts:21]: what about the slippage and fee configuration, how that could be controlled from the caller?

**Comment by paszkowskiDamian** [comment]
Looks good for me however I wanted to point out that we didn't think about integrating summer fees in the swap service.

**Review by paszkowskiDamian — DISMISSED** [review]

**Review by piotrwitek — APPROVED** [review]
looks great

**Review by piotrwitek — APPROVED** [review]

**Pull request merged**